runtime.go

```diff
@@ -1992,11 +1992,17 @@ func (r *Runtime) wrapReflectFunc(value reflect.Value) func(FunctionCall) Value
 
 		if last := out[len(out)-1]; last.Type().Name() == "error" {
 			if !last.IsNil() {
-				err := last.Interface()
+				err := last.Interface().(error)
 				if _, ok := err.(*Exception); ok {
 					panic(err)
 				}
-				panic(r.NewGoError(last.Interface().(error)))
+				var intErr *InterruptedError
+				if errors.As(err, &intErr) {
```

Comment:
I think a simple type assertion would be enough here, the *InterruptedError is never going to be wrapped by goja, and if wrapped by the user, well... they've asked for it:)

Reply:
I would expect that in some cases users will just wrap w/e error they get and add some more info to it, which IIRC is what is currently recommended. 

In that case they will now need to be careful that it is not `InterruptedError` and if it is to *not* wrap this one specifically. 

I am not certain how much worse `errors.As` is compared to a cast, but I would expect it will be negligible compared to anything else goja needs to do with reflect in this part of the code alone.

Reply:
Fair point. But, thinking about it, a more generic solution is needed here because InterruptedError is not the only type that is thrown as an uncatchableException. I'll look into this.